src/ext/draft/context_menu.rs

```diff
@@ -0,0 +1,138 @@
+    pub is_enabled: bool,
+    pub action_id: clap_id,
+}
+
+unsafe impl Send for clap_context_menu_entry {}
+unsafe impl Sync for clap_context_menu_entry {}
+
+#[repr(C)]
+#[derive(Debug, Copy, Clone)]
+pub struct clap_context_menu_check_entry {
+    pub label: *const c_char,
+    pub is_enabled: bool,
+    pub is_checked: bool,
+    pub action_id: clap_id,
+}
+
+unsafe impl Send for clap_context_menu_check_entry {}
+unsafe impl Sync for clap_context_menu_check_entry {}
+
+#[repr(C)]
+#[derive(Debug, Copy, Clone)]
+pub struct clap_context_menu_item_title {
+    pub title: *const c_char,
+    pub is_enabled: bool,
+}
+
+unsafe impl Send for clap_context_menu_item_title {}
+unsafe impl Sync for clap_context_menu_item_title {}
+
+#[repr(C)]
+#[derive(Debug, Copy, Clone)]
+pub struct clap_context_menu_submenu {
+    pub label: *const c_char,
+    pub is_enabled: bool,
+}
+
+unsafe impl Send for clap_context_menu_submenu {}
+unsafe impl Sync for clap_context_menu_submenu {}
+
+#[repr(C)]
+#[derive(Debug, Copy, Clone)]
+pub struct clap_context_menu_builder {
+    pub ctx: *mut c_void,
+    pub add_item: Option<
+        unsafe extern "C" fn(
+            builder: *const clap_context_menu_builder,
+            item_kind: clap_context_menu_item_kind,
+            item_data: *const c_void,
+        ) -> bool,
+    >,
+    pub supports: Option<
+        unsafe extern "C" fn(
+            builder: *const clap_context_menu_builder,
+            item_kind: clap_context_menu_item_kind,
+        ) -> bool,
+    >,
+}
+
+unsafe impl Send for clap_context_menu_builder {}
+unsafe impl Sync for clap_context_menu_builder {}
+
+#[repr(C)]
+#[derive(Debug, Copy, Clone)]
+pub struct clap_plugin_context_menu {
+    pub populate: Option<
+        unsafe extern "C" fn(
+            plugin: *const clap_plugin,
+            target: *const clap_context_menu_target,
+            builder: *const clap_context_menu_builder,
+        ) -> bool,
+    >,
+    pub perform: Option<
+        unsafe extern "C" fn(
+            plugin: *const clap_plugin,
+            target: *const clap_context_menu_target,
+            action_id: clap_id,
+        ) -> bool,
+    >,
+}
+
+#[repr(C)]
+#[derive(Debug, Copy, Clone)]
+pub struct clap_host_context_menu {
+    pub populate: Option<
+        unsafe extern "C" fn(
+            host: *const clap_host,
+            target: *const clap_context_menu_target,
+            builder: *const clap_context_menu_builder,
+        ) -> bool,
+    >,
+    pub perform: Option<
+        unsafe extern "C" fn(
+            host: *const clap_host,
+            target: *const clap_context_menu_target,
+            action_id: clap_id,
+        ) -> bool,
+    >,
+    pub can_popup: Option<unsafe extern "C" fn(host: *const clap_host) -> bool>,
+    pub popup: Option<
+        unsafe extern "C" fn(
+            host: *const clap_host,
+            target: *const clap_context_menu_target,
+            screen_index: u32,
+            x: u32,
+            y: u32,
```

Comment:
These should be `i32`: https://github.com/free-audio/clap/blob/065d685d4e9657f0344f350eef748be2b4d8e318/include/clap/ext/draft/context-menu.h#L151-L153